src/HLSLTree.cpp

```diff
@@ -382,6 +382,40 @@ bool HLSLTree::GetExpressionValue(HLSLExpression * expression, int & value)
     return false;
 }
 
+bool HLSLTree::NeedsFunction(const char* name)
+{
+    // Early out
+    if (!GetContainsString(name))
+        return false;
+
+    struct NeedsFunctionVisitor: HLSLTreeVisitor
```

Comment:
My understanding is that the visitor infrastructure doesn't make it too easy. I'll check how much time NeedsFunction takes - it may be worthwhile to skip branches in smth like VisitTopLevelStatement if we already have found our function, although of course for failed lookups - which are probably more common! - it's not beneficial to early out.